src/styles/animations.js

```diff
@@ -0,0 +1,69 @@
+var AutoPrefix = require('./auto-prefix');
+
+var createdAnimations = {};
+
+function createSheet(){
+
+	var styleElement = document.createElement("style");
+	document.head.appendChild(styleElement);
+	var sheet = styleElement.sheet;
+
+	return sheet;
+}
+
+function addRule(sheet, selector){
+	var size = (sheet.cssRules || sheet.rules).length;
+	if("insertRule" in sheet) {
+		sheet.insertRule(selector + " { }", size);
+	}
+	else if("appendRule" in sheet){
+		sheet.appendRule(selector + " { }");
+	}
+	else if("addRule" in sheet) {
+		sheet.addRule(selector, " {} ", size);
+	}
+}
+ 
+function setStyles(rule, obj){
+	for(var prop in obj){
+		rule.style[prop] = obj[prop];
+	}
+}
+
+module.exports = {
+
+	 create: function (name, frames){
+
+	 	if(name in createdAnimations){
+	 		return name;
+	 	}
+	 	
+	 	createdAnimations[name] = true;
+
+		var sheet = createSheet();
+		 
+		var keyframeTags = [ "@keyframes ", "@-webkit-keyframes "];
```

Comment:
@castrolol The performance loss is most likely due applying transition to width and not because of setTimeout. I would recommend using transform scale3D instead. See:
http://blogs.adobe.com/webplatform/2014/03/18/css-animations-and-transitions-performance/